symengine/dense_matrix.cpp

```diff
@@ -71,7 +71,7 @@ bool DenseMatrix::is_lower() const
     unsigned n = A.nrows();
     for (unsigned i = 1; i < n; ++i) {
         for (unsigned j = 0; j < i; ++j) {
-            if (not is_zero(*A.get(i, j))) {
+            if (not is_true(is_zero(*A.get(i, j)))) {
```

Comment:
I think we should use `is_number_and_zero` here.
```suggestion
            if (not is_number_and_zero(*A.get(i, j))) {
```

Reply:
Changed to `is_number_and_zero`.